openbb_platform/obbject_extensions/charting/openbb_charting/query_params.py

```diff
@@ -1,47 +1,167 @@
 """Charting Extension Query Params."""
 
-from typing import List, Optional, Union
+from typing import Any, Dict, List, Literal, Optional, Union
 
 from openbb_core.provider.abstract.data import Data
 from openbb_core.provider.abstract.query_params import QueryParams
 from pydantic import Field
 
+from openbb_charting.core.to_chart import ChartIndicators
```

Comment:
I love the idea of having everything really explicit and as pydantic models!
But, I believe the chain of inheritance could be a lot simplified, example:

`BaseModel -> QueryParams -> BaseQueryParams -> IndicatorsQueryParams -> MAIndicatorsQueryParams -> SMAIndicatorsQueryParams`

`BaseModel -> QueryParams -> BaseQueryParams -> ChartQueryParams -> EquityPricePerformanceChartQueryParams -> EtfPricePerformanceChartQueryParams`

> imo, bad idea to mix equity and etf things, as discussed in previous situations

Do we need 4/5 levels of inheritance?

As a suggestion, I think this could be enough (1/2 levels of inheritance):

`BaseModel -> MAIndicatorsQueryParams -> SMAIndicatorsQueryParams`

Reply:
Inheriting from BaseModel is not going to cut it here.

- `EquityPricePerformanceChartQueryParams -> EtfPricePerformanceChartQueryParam`
  - This is exactly what needs to happen. For all purposes, there is no difference between the two other than getting a new name and docstring.

`ChartParams` and `IndicatorsQueryParams` do not have the same fields but are both children of QueryParams. 

`BaseQueryParams` exists so that the `__repr__` method is inherited everywhere I need it.

`IndicatorsParams(QueryParams)` exists like this because it should not inherit from `BaseQueryParams`.